keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,461 @@ non-goals of this KEP.
-
-	// Attributes defines the set of attributes for this device.
-	// The name of each attribute must be unique in that set.
-	//
-	// To ensure this uniqueness, attributes defined by the vendor
-	// must be listed without the driver name as domain prefix in
-	// their name. All others must be listed with their domain prefix.
-	//
-	// Conflicting attributes from those provided via mixins are
-	// overwritten by the ones provided here.
-	//
-	// The maximum number of attributes and capacities combined is 32.
-	//
-	// +optional
-	Attributes map[QualifiedName]DeviceAttribute `json:"attributes,omitempty"`
-
-	// Capacity defines the set of capacities for this device.
-	// The name of each capacity must be unique in that set.
-	//
-	// To ensure this uniqueness, capacities defined by the vendor
-	// must be listed without the driver name as domain prefix in
-	// their name. All others must be listed with their domain prefix.
-	//
-	// Conflicting capacities from those provided via mixins are
-	// overwritten by the ones provided here.
-	//
-	// The maximum number of attributes and capacities combined is 32.
-	//
-	// +optional
-	Capacity map[QualifiedName]DeviceCapacity `json:"capacity,omitempty"`
+  // Includes defines the set of device mixins that this device includes.
+  //
+  // The propertes of each included mixin are applied to this device in
+  // order. Conflicting properties from multiple mixins are taken from the
+  // last mixin listed that contains them. Properties set on the device will
+  // always override properties from mixins.
+  //
+  // The mixins referenced here must be defined in the same
+  // ResourceSlice.
+  //
+  // The maximum number of mixins that can be included is 8.
+  //
+  // +optional
+  // +listType=atomic
+  Includes []DeviceMixinRef
+
+  // Attributes defines the set of attributes for this device.
+  // The name of each attribute must be unique in that set.
+  //
+  // To ensure this uniqueness, attributes defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting attributes from those provided via mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of attributes and capacities combined is 32.
+  //
+  // +optional
+  Attributes map[QualifiedName]DeviceAttribute
+
+  // Capacity defines the set of capacities for this mixin.
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting capacities from those provided via other mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of attributes and capacities combined is 32.
+  //
+  // +optional
+  Capacity map[QualifiedName]DeviceCapacity
```

Comment:
Does it make sense to rename this `AdvertisedCapacity` to contrast it with `ConsumedCapacity`?

Reply:
Yeah, I did think about a more descriptive name here. I think it makes sense, but I am not sure if this makes the mapping from devices in the `ResourceSlice` to the properties of the CEL expressions in the `ResourceClaims` less obvious? 
I was also uncertain about introducing differences in naming between `BasicDevice` and `CompositeDevice`, although that would be something we can fix in `v1beta2`.

Reply:
My $0.02 - it's fine as is.